src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs

```diff
@@ -100,6 +123,36 @@ internal static string GetAccountId(string functionArn)
             return null;
         }
 
+        internal static IEnumerable<KeyValuePair<string, object>> GetFunctionDefaultTags<TInput>(TInput input, ILambdaContext context)
+        {
+            var trigger = "other";
+            if (input is APIGatewayProxyRequest || input is APIGatewayHttpApiV2ProxyRequest)
+            {
+                trigger = "http";
+            }
+
+            var tags = new List<KeyValuePair<string, object>>
+            {
+                new(AWSLambdaSemanticConventions.AttributeFaasTrigger, trigger),
+            };
+
+            if (context == null)
+            {
+                return tags;
+            }
+
+            tags.Add(new(AWSLambdaSemanticConventions.AttributeFaasName, context.FunctionName));
+            tags.Add(new(AWSLambdaSemanticConventions.AttributeFaasID, context.InvokedFunctionArn));
+
+            var functionParts = context.InvokedFunctionArn?.Split(':');
+            if (functionParts != null && functionParts.Length >= 5)
+            {
+                tags.Add(new(AWSLambdaSemanticConventions.AttributeCloudAccountID, functionParts[4]));
```

Comment:
It seems this is already set directly in OnFunctionStart and is redundant here.

Reply:
All moved to AWSLambdaUtils.GetFunctionDefaultTags(...) method.